### PR TITLE
Ensures story video has sources before playing to avoid early error.

### PR DESCRIPTION
**Pull request by gmajoulet**
TLDR:
https://github.com/ampproject/amphtml/pull/23794 introduced better error handling for story videos. This new code will display the error message without trying to play the video if it has no sources.
Calling `preloadMedia_` before `playMedia_` ensures sources are set by the media pool, and avoids the early return.

More context:
This isn't the usual code path (and I didn't know this one existed tbh), and only happens when the `amp-video` has no height (`resource.isDisplayed()` returns false) when we first try to play it. In this case, we don't set the video sources to make sure nothing loads, which saves users data. A few frames later the video is detected as visible, and we try to play it. This PR ensures that, in this very case, the sources are added back before trying to play it.

Fixes https://github.com/ampproject/amphtml/issues/24414

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story-page.js [hunk 1]
```diff
@@ -1372,12 +1383,14 @@ export class AmpStoryPage extends AMP.BaseElement {
...
     this.mediaPoolPromise_.then(mediaPool => {
       if (visible) {
-        this.registerMedia_(mediaPool, videoEl).then(() => {
-          this.playMedia_(mediaPool, videoEl);
-          if (!this.storeService_.get(StateProperty.MUTED_STATE)) {
-            this.unmuteAllMedia();
-          }
-        });
+        this.registerMedia_(mediaPool, videoEl)
+          .then(() => this.preloadMedia_(mediaPool, videoEl))
```
Comment (choumx) [inline extensions/amp-story/1.0/amp-story-page.js]: Should this be done in `playMedia_()` instead?

Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story-page.js]: Having this in two steps (preload / play) is convenient for two main reasons:
- it keeps `playMedia_` lighter and optimizes play/pause interactions, like hold to pause
- it allows us to start listening for media events (waiting/playing/...) between the preload and the play actions, to avoid extra events sent when updating the sources

Pretty sure we can find workarounds if you feel strongly though. The first point can be fixed by making `preloadMedia_` a noop when needed, and the second one by handling these events differently

Comment (choumx) [reply extensions/amp-story/1.0/amp-story-page.js]: Sure, just curious if we should enforce the invariant in `playMedia_()` instead of relying on callers to know what to do.
